### ISSUE TITLE
Auto compaction tasks leaks disk space in case of failure

### ISSUE THREAD

**Issue by RestfulBlue** [apache/incubator-druid#7505]
Hi, recently we was trying to use auto compaction, but found some strange behavior of temporary data location. At first we make a mistake and use java.io.tmpdir in tmpfs. After some time total amount of compaction data become 50gb. 

It looks like at some time there was no enough ram to fit all the things and tasks starts failing. But with that it looks like after task was failed space was not released. Even when no tasks is running, we was seeing about 48gb of data holded in druid jvm tmp dir. 

After that we disable auto compaction and switch druid tmp dir to somethere else. But auto compaction is really important for us. for example we have tables with 3 thousands of segments in tables with 1gb total :(

In other words, it looks like compaction tasks leaks disk space in case of failure. Is there a location, where we can point java.io.tmpdir and be sure space will not be leaked?

**Comment by gianm** [comment]
Hmm, it seems like a bug in the indexing task. Each task has a working directory that it's supposed to use for temporary data, and which will get cleaned up if the task fails. Did you see what kind of data shows up in java.io.tmpdir? Should help give a clue as to what's generating it - and then that could be moved to the task working directory.

**Comment by RestfulBlue** [comment]
Hi, thanks for response! in our configuration we was settings baseTaskDir and not baseDir. And it looks like last 2 years all tasks data was storing in tmpfs :} . Overflow of tmpfs was leading to node OOM and probably that cause some unexpected behaviour. That was not a problem before because we have a lot of space in tmpfs and we was running only kafka indexing tasks.  We will change all locations to another place and if problem raise again i will reopen this issue. Sorry for worriyng!
